src/platform/unix/mod.rs

```diff
@@ -465,19 +464,23 @@ impl OsIpcReceiverSet {
                                 shared_memory_regions));
                     }
                     Err(err) if err.channel_is_closed() => {
-                        hangups.insert(pollfd.fd);
-                        selection_results.push(OsIpcSelectionResult::ChannelClosed(
-                                    *self.fdids.get(&pollfd.fd).unwrap()))
+                        let id = self.fdids.remove(&pollfd.fd).unwrap();
+                        unsafe {
+                            libc::close(pollfd.fd);
+                        }
+                        selection_results.push(OsIpcSelectionResult::ChannelClosed(id))
                     }
                     Err(err) => return Err(err),
                 }
                 pollfd.revents = pollfd.revents & !POLLIN
             }
         }
 
-        if !hangups.is_empty() {
-            self.pollfds.retain(|pollfd| !hangups.contains(&pollfd.fd));
-        }
+        // Avoid the use of `self` in closue to avoid errors due to the
+        // mutable borrow of `self` at line 483
+        let ids = &self.fdids;
```

Comment:
> (BTW, including line numbers in a comment is a bad idea: these are bound to change :-) )

Ha true! good call... Removed the line numbers

I also added a comment above the retain detailing the reason for the placement.